### DISCUSSION TITLE
Permissions

### DISCUSSION THREAD

**Discussion by kaaax0815** [wg-easy/wg-easy#1353] [Q&A]
@pheiduck @tetuaoro 
#1333 Looks very good the next big steps are better UI (Client Creation, Editing (to support new stuff like Allowed IPs, IPv6)
and Permissions.

We have currently 3 roles. `ADMIN`, `EDITOR` and `CLIENT`, they are not currently working but its a start. A requested feature is groups. But how do we implement them? How should the permission system look like?

I thought a user should be able to create a group and add multiple clients to it. Then be able to assign that group to one (or more?) users. If that user is a editor he should be able to add clients. If he's a client he can only download the config etc.

There is also a scenario where admins, create a user for a person and that person then should be able to create their own clients. Which wouldn't be supported with above implementation

**Comment by tetuaoro** [comment]
> I thought a user should be able to create a group and add multiple clients to it. Then be able to assign that group to one (or more?) users. If that user is a editor he should be able to add clients. If he's a client he can only download the config etc

Clients will be able to download his configuration and an `EDITOR` to manage wg server. Sounds good 👍.

Will a client also be able to log in to the server ? What do you thing about ?

**Reply by tetuaoro** [reply]
I mean, that's why I introduced `ROLE`.

**Reply by kaaax0815** [reply]
Clients can login. See all their assigned group(s) and clients (peers) They can download them (maybe edit name?) but not address etc.

**Comment by pheiduck** [comment]
> But how do we implement them? How should the permission system look like?

ADMIN: should be able to delete, edit, create, enable, disable (all) client configs
EDITOR: should be able to edit, enable, disable (all) client configs
CLIENT: should be able to view, enable, disable (all) client configs

**Reply by kaaax0815** [reply]
I wanted to include 2 features:
>User groups
> Implement user grouping to manage permissions, such as enabling or disabling internet access for specific groups.

and 

> multiple Users that are able to login and create e.g. upto x peers

**Comment by tetuaoro** [comment]
`EDITOR`s manages clients and `ADMIN`s manages server and clients. What do you think ?!

**Reply by kaaax0815** [reply]
Editors and Clients only see what everyone sees currently (`master`). Admins will see the new settings page. Thats what I was going for
